packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -1707,6 +1707,74 @@ void main() {
       );
     });
 
+    // Regression test for https://github.com/flutter/flutter/issues/159942.
+    testWidgets('OutlineBorder does not overlap with the label when there is no radius', (WidgetTester tester) async {
+      Widget buildFrame(TextDirection textDirection) {
+        return MaterialApp(
+          home: Scaffold(
+            body: Container(
+              padding: const EdgeInsets.all(16.0),
+              alignment: Alignment.center,
+              child: Directionality(
+                textDirection: textDirection,
+                child: const RepaintBoundary(
+                  child: InputDecorator(
+                    isFocused: true,
+                    decoration: InputDecoration(
+                      labelText: labelText,
+                      border: OutlineInputBorder(
+                        gapPadding: 0.0,
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        );
+      }
+
+      await tester.pumpWidget(buildFrame(TextDirection.ltr));
+      Rect labelRect = getLabelRect(tester);
+      RenderBox borderBox = InputDecorator.containerOf(tester.element(findBorderPainter()))!;
+      expect(findBorderPainter(), paints
+        ..save()
+        ..path(
+          // The points of the label edge should be part of the border.
+          includes: <Offset>[
+            borderBox.globalToLocal(labelRect.centerLeft),
+            borderBox.globalToLocal(labelRect.centerRight),
+          ],
+          // The points inside the label should not be part of the border.
+          excludes: <Offset>[
+            borderBox.globalToLocal(labelRect.centerLeft) + const Offset(1, 0),
+            borderBox.globalToLocal(labelRect.centerRight) + const Offset(-1, 0),
+          ],
```

Comment:
The reason there are no variables stored here is to enhance the readability of the code. If needed, I will store them as `labelCenterLeftLocalToBorder` and `labelCenterRightLocalToBorder`.